vllm/core/block/naive_block.py

```diff
@@ -0,0 +1,271 @@
+from typing import Dict, Iterable, List, Optional, Set
+
+from vllm.core.block.common import (CopyOnWriteTracker, RefCounter,
+                                    get_all_blocks_recursively)
+from vllm.core.block.interfaces import Block, BlockAllocator
+
+BlockId = int
+Refcount = int
+
+
+class NaiveBlockAllocator(BlockAllocator):
```

Comment:
I think this should be called CoWBlockAllocator btw because it's actually quite powerful

Reply:
I thought about this -- the conflict is that the prefix caching allocator also supports CoW (it isn't unique to this allocator).